### PR TITLE
Use specific miniconda tag, do not perform apt-get upgrade, combine conda steps

### PR DESCRIPTION
**Pull request by isuftin**

### CODE REVIEW

File: nhmusgs-base/Dockerfile [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
+FROM continuumio/miniconda3:4.7.12
 
 LABEL maintainer="ashalper@usgs.gov"
 
-# ...check for any Debian, installed packages updates, upgrade them and install
-# additional packages
-RUN apt-get update && apt-get -y upgrade && \
+# install additional packages
+RUN apt-get update && \
     apt-get -y install dialog file gcc gfortran make procps unzip && \
```
Comment (isuftin) [inline nhmusgs-base/Dockerfile]: Why do we install gcc and gfortran and make here? Is this used by multiple images using this base image? Can we push these packages down to the image that actually uses it?

**Comment by isuftin** [comment]
This update shaves down the base image from 3.69GB to 3.08GB

**Comment by isuftin** [comment]
The miniconda base image is 443MB, which means we are packing a lot of stuff into this base image. It looks like the apt-get command still creates a 210MB layer and the conda command set creates  a 2.49GB layer.

File: nhmusgs-base/Dockerfile (hunk 1 above)
Comment (ashalper-usgs) [reply nhmusgs-base/Dockerfile]: Yes. I'm pretty sure it's only used by `nhm-prms` to build PRMS. Should I do that, or do you want to?

**Comment by ashalper-usgs** [comment]
Yes, not surprising. I've noticed just from watching the Conda update process that it seems like there are redundant packages between Conda and Debian (although they are probably different versions). Conda seems to (in part, or in full) ignore the packages installed on the host system.

**Pull request merged**